barretenberg/cpp/src/barretenberg/bb/api_client_ivc.hpp

```diff
@@ -0,0 +1,266 @@
+        vinfo("writing ClientIVC proof and vk...");
+        write_file(output_dir / "client_ivc_proof", to_buffer(proof));
+
+        auto eccvm_vk = std::make_shared<ECCVMFlavor::VerificationKey>(ivc.goblin.get_eccvm_proving_key());
+        auto translator_vk =
+            std::make_shared<TranslatorFlavor::VerificationKey>(ivc.goblin.get_translator_proving_key());
+        write_file(output_dir / "client_ivc_vk",
+                   to_buffer(ClientIVC::VerificationKey{ ivc.honk_vk, eccvm_vk, translator_vk }));
+    };
+
+    /**
+     * @brief Verifies a client ivc proof and writes the result to stdout
+     *
+     * Communication:
+     * - proc_exit: A boolean value is returned indicating whether the proof is valid.
+     *   an exit code of 0 will be returned for success and 1 for failure.
+     *
+     * @param proof_path Path to the file containing the serialized proof
+     * @param vk_path Path to the serialized verification key of the final (MegaHonk) circuit in the stack
+     * @param accumualtor_path Path to the file containing the serialized protogalaxy accumulator
+     * @return true (resp., false) if the proof is valid (resp., invalid).
+     */
+    bool verify([[maybe_unused]] const API::Flags& flags,
+                const std::filesystem::path& proof_path,
+                const std::filesystem::path& vk_path) override
+    {
+        // TODO(https://github.com/AztecProtocol/barretenberg/issues/1163): Set these dynamically
+        init_bn254_crs(1);
+        init_grumpkin_crs(1 << 15);
+
+        const auto proof = from_buffer<ClientIVC::Proof>(read_file(proof_path));
+        const auto vk = from_buffer<ClientIVC::VerificationKey>(read_file(vk_path));
+
+        vk.mega->pcs_verification_key = std::make_shared<VerifierCommitmentKey<curve::BN254>>();
+        vk.eccvm->pcs_verification_key =
+            std::make_shared<VerifierCommitmentKey<curve::Grumpkin>>(vk.eccvm->circuit_size + 1);
+        vk.translator->pcs_verification_key = std::make_shared<VerifierCommitmentKey<curve::BN254>>();
+
+        const bool verified = ClientIVC::verify(proof, vk);
+        vinfo("verified: ", verified);
+        return verified;
+    };
+
+    bool prove_and_verify(const API::Flags& flags,
+                          const std::filesystem::path& bytecode_path,
+                          const std::filesystem::path& witness_path) override
+    {
+        if (!flags.input_type || !(*flags.input_type == "compiletime_stack" || *flags.input_type == "runtime_stack")) {
+            throw_or_abort("No input_type or input_type not supported");
+        }
+        std::vector<acir_format::AcirProgram> folding_stack =
+            _build_folding_stack(*flags.input_type, bytecode_path, witness_path);
+        ClientIVC ivc = _accumulate(folding_stack);
+        const bool verified = ivc.prove_and_verify();
+        return verified;
+    };
+
+    void gates([[maybe_unused]] const API::Flags& flags,
```

Comment:
add a TODO for this three functions

Reply:
or actually I guess they are not client ivc related but as a next step we should create an ultra honk api class as well?